File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        unordered_map<string,function<int(int,int)>> ops =  {{"+", [](int x, int y){ return x+y;}},
                                                             {"-", [](int x, int y){ return x-y;}},
                                                             {"*", [](int x, int y){ return x*y;}},
                                                             {"/", [](int x, int y){ return x/y;}}};
        stack<int> st;
        for (const string& token : tokens){
            if (ops.contains(token)){
                int y = st.top();
                st.pop();
                int x = st.top();
                st.pop();
                st.push(ops[token](x,y));
            } else 
                st.push(stoi(token));
        }
        return st.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        ops = {"+" : lambda x,y: x+y,
               "-" : lambda x,y: x-y,
               "*" : lambda x,y: x*y,
               "/" : lambda x,y: floor(x/y) if x*y > 0 else ceil(x/y)}
        st = deque()
        for token in tokens:
            if token in ops:
                n2 = st.pop()
                n1 = st.pop()
                st.append(ops[token](n1,n2))
            else:
                st.append(int(token))
        return st[0]